extensions/amp-analytics/0.1/session-manager.js

```diff
@@ -106,48 +98,64 @@ export class SessionManager {
       hasOwn(this.sessions_, type) &&
       !isSessionExpired(this.sessions_[type])
     ) {
+      this.sessions_[type] = this.updateSession_(this.sessions_[type]);
       this.setSession_(type, this.sessions_[type]);
-      this.sessions_[type].lastAccessTimestamp = Date.now();
       return Promise.resolve(this.sessions_[type]);
     }
 
-    return this.getSessionFromStorage_(type);
+    return this.getOrCreateSession_(type);
   }
 
   /**
    * Get our session if it exists or creates it. Sets the session
-   * in localStorage to update the last access time.
+   * in localStorage to update the access time.
    * @param {string} type
    * @return {!Promise<SessionInfoDef>}
    */
-  getSessionFromStorage_(type) {
+  getOrCreateSession_(type) {
     return this.storagePromise_
       .then((storage) => {
         const storageKey = getStorageKey(type);
-        // Gets the session if it has not expired
-        return storage.get(storageKey, SESSION_MAX_AGE_MILLIS);
+        return storage.get(storageKey);
       })
       .then((session) => {
-        // If no valid session in storage, create a new one
+        // Either create session or update it
         return !session
-          ? constructSessionInfo(generateSessionId(), Date.now())
-          : constructSessionFromStoredValue(session);
+          ? constructSessionInfo()
+          : this.updateSession_(constructSessionFromStoredValue(session));
       })
       .then((session) => {
         // Avoid multiple session creation race
         if (type in this.sessions_ && !isSessionExpired(this.sessions_[type])) {
           return this.sessions_[type];
         }
         this.setSession_(type, session);
-        session.lastAccessTimestamp = Date.now();
         this.sessions_[type] = session;
         return this.sessions_[type];
       });
   }
 
+  /**
+   * Check if session has expired and reset values (id, count) if so.
+   * Also update `accessTimestamp`.
+   * @param {!SessionInfoDef} session
+   * @return {!SessionInfoDef}
+   */
+  updateSession_(session) {
+    const currentCount = session[SESSION_VALUES.COUNT];
+    if (isSessionExpired(session)) {
+      const newSessionCount = currentCount === undefined ? 1 : currentCount + 1;
```

Comment:
readability suggestion:
```suggestion
      const newSessionCount = (currentCount ?? 0) + 1;
```